Sources/AppStoreConnectCLI/Helpers/Publisher+Helpers.swift

```diff
@@ -3,20 +3,72 @@
 import Combine
 import Foundation
 
-extension Publisher where Output: ResultRenderable, Failure == Error {
-    func renderResult(format: OutputFormat) -> AnyCancellable {
-        self.sink(
-            receiveCompletion: Renderers.CompletionRenderer().render,
-            receiveValue: Renderers.ResultRenderer<Output>(format: format).render
-        )
+enum PublisherAwaitError: LocalizedError {
+    case timedOut(timeout: DispatchTime)
+    case expectedOutput
+    case expectedSingleOutput(outputCount: Int)
+
+    var failureReason: String? {
+        switch self {
+        case .timedOut(let timeout):
+            return "Expected publisher output but timed out with timeout: \(timeout)"
+        case .expectedOutput:
+            return "Expected publisher output but none received"
+        case .expectedSingleOutput(let outputCount):
+            return "Expected single publisher output but received \(outputCount)"
+        }
     }
 }
 
-extension Publisher where Output == Void, Failure == Error {
-    func renderResult(format: OutputFormat) -> AnyCancellable {
-        self.sink(
-            receiveCompletion: Renderers.CompletionRenderer().render,
-            receiveValue: Renderers.null
-        )
+extension Publisher {
+    func await(timeout: DispatchTime = .now() + 30) throws -> Output {
```

Comment:
It seems `timeout` isn't being used in this function?
```suggestion
    func await() throws -> Output {
```

Reply:
Oh the timeout should be passed to the `awaitMany` function 😂 
I think I'll change the default to `distantFuture`